src/buffer.cc

```diff
@@ -30,16 +33,104 @@ FormatBuffer* Buffer::AsFormatBuffer() {
   return static_cast<FormatBuffer*>(this);
 }
 
+Result Buffer::CopyBaseFieldsTo(Buffer* buffer) const {
+  if (buffer->width_ != width_)
+    return Result("Buffer::CopyBaseFields() buffers have a different width");
+  if (buffer->height_ != height_)
+    return Result("Buffer::CopyBaseFields() buffers have a different height");
+  if (buffer->size_ != size_)
+    return Result("Buffer::CopyBaseFields() buffers have a different size");
+  buffer->data_.clear();
+  buffer->data_ = data_;
+  buffer->values_.clear();
+  buffer->values_ = values_;
+  return {};
+}
+
+Result Buffer::CopyTo(Buffer* buffer) const {
+  Result result = CopyBaseFieldsTo(buffer);
+  if (!result.IsSuccess())
+    return result;
+
+  if (buffer->mem_ptr_) {
+    // TODO(hevrard) Return error if buffer_to mem_ptr_ is backed by GPU memory
+    std::free(buffer->mem_ptr_);
```

Comment:
You can remove the if wrapper, it's safe to call free on a nullptr.

Reply:
OK, I wasn't sure what is the best practice for Amber.